cmd/kubeadm/app/util/config/masterconfig.go

```diff
@@ -43,12 +43,17 @@ func SetInitDynamicDefaults(cfg *kubeadmapi.MasterConfiguration) error {
 	cfg.API.AdvertiseAddress = ip.String()
 
 	// Validate version argument
-	ver, err := kubeadmutil.KubernetesReleaseVersion(cfg.KubernetesVersion)
+	ver, useCI, err := kubeadmutil.KubernetesValidateVersion(cfg.KubernetesVersion)
 	if err != nil {
 		return err
 	}
 	cfg.KubernetesVersion = ver
 
+	// Requested version is automatic CI build, thus use KubernetesCI Image Repository for core images
+	if useCI {
```

Comment:
Why not just move this up to check `strings.HasPrefix(cfg.KubernetesVersion, "ci/")` and set the CI repo on that condition? KubernetesValidateVersion will be much simpler then...